overlord/snapstate/snapstate.go

```diff
@@ -1151,7 +1189,11 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		}
 	}
 
-	newAutoAliases, mustPruneAutoAliases, transferTargets, err := autoAliasesUpdate(st, names, updates)
+	toUpdate := make([]minimalInstallInfo, len(updates))
+	for i, up := range updates {
+		toUpdate[i] = &manualUpdateInfo{*up}
+	}
```

Comment:
I'm not a fan of all this copying,  I know we discussed not to do that but now that we are looking at just this set of changes, it's not clear to me that there shouldn't be an easy way to make doUpdate take directly  []mininalInstallInfo.

I'm happy to chat about this